### PR TITLE
Icmp Extensions v2

### LINKED ISSUE #751: Add support for parsing ICMP extensions in IPv4/IPv6

### LINKED ISSUE #753: Add support for MPLS Label Stack Class ICMP Extension Object

### LINKED ISSUE #807: Add support for ICMP Extensions in tracing backend

### PR DESCRIPTION
**Pull request by fujiapple852**
Closes #751 
Closes #753 
Closes #807

### CODE REVIEW

File: src/tracing/probe.rs [hunk 1]
```diff
@@ -128,17 +128,46 @@ pub enum IcmpPacketType {
...
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct MplsLabelStackMember {
+    pub label: u32,
+    pub exp: u8,
+    pub bos: u8,
+    pub ttl: u8,
+}
+
 /// The data in the probe response.
-#[derive(Debug, Copy, Clone)]
```
Comment (fujiapple852) [inline src/tracing/probe.rs:141]: In wrong commit

File: src/config.rs [hunk 2]
```diff
@@ -363,6 +364,13 @@ impl TryFrom<(Args, &Platform)> for TrippyConfig {
             cfg_file_strategy.tos,
             constants::DEFAULT_STRATEGY_TOS,
         );
+
+        let icmp_extensions = cfg_layer_bool_flag(
+            args.icmp_extensions,
+            cfg_file_strategy.icmp_extensions,
+            false,
+        );
+
```
Comment (fujiapple852) [inline src/config.rs:373]: Spaces here

File: trippy-config-sample.toml [hunk 3]
```diff
@@ -160,6 +160,18 @@ payload-pattern = 0
 # This is also known as DSCP+ECN.
 tos = 0
 
+# Whether to parse ICMP extensions.
+#
+# If enabled, all extensions attached to incoming ICMP TimeExceeded and DestinationUnavailable messages will be parsed
+# and proved as part of the trace response data.
```
Comment (fujiapple852) [inline trippy-config-sample.toml:166]: Typo

File: src/tracing/packet/icmp_extension.rs [hunk 4]
```diff
@@ -0,0 +1,1053 @@
...
+                0x20, 0x00, 0x99, 0x3a, 0x00, 0x08, 0x01, 0x01, 0x04, 0xbb, 0x41, 0x01,
+            ];
+            let extensions = ExtensionsPacket::new_view(&buf).unwrap();
+            let header = ExtensionHeaderPacket::new_view(extensions.header()).unwrap();
+            assert_eq!(2, header.get_version());
+            assert_eq!(0x993A, header.get_checksum());
+        }
+
+        #[test]
+        fn test_object_iterator() {
```
Comment (fujiapple852) [inline src/tracing/packet/icmp_extension.rs:116]: Add test for malformed object with zero length

File: src/tracing/packet/icmp_extension.rs [hunk 5]
```diff
@@ -0,0 +1,1053 @@
...
+                offset: 0,
+                bos: 0,
+            }
+        }
+    }
+
+    impl<'a> Iterator for MplsLabelStackIter<'a> {
+        type Item = &'a [u8];
+
+        fn next(&mut self) -> Option<Self::Item> {
```
Comment (fujiapple852) [inline src/tracing/packet/icmp_extension.rs:557]: Review termination cases here when malformed packets

File: src/tracing/packet/icmp_extension.rs [hunk 6]
```diff
@@ -0,0 +1,1053 @@
...
+                "
+               0b 00 f4 ee 00 11 00 00 45 00 00 54 20 c3 40 00
+               02 01 b5 7e 64 63 08 2a 5d b8 d8 22 08 00 11 8d
+               65 83 80 ef 00 00 00 00 00 00 00 00 00 00 00 00
+               00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+               00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+               00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+               00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+               00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+               00 00 00 00 00 00 00 00 20 00 78 56 00 08 01 01
```
Comment (fujiapple852) [inline src/tracing/packet/icmp_extension.rs:1043]: Looks like a possible extension object?
